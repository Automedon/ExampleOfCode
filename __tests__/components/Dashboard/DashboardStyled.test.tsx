import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import {
  CarInStock,
  Row,
  DisplayCarsGrid,
  FormWrapper,
  StyledSelect,
} from "../../../components/Dashboard/DashboardStyled";

describe("Styled-components Dashboard", () => {
  it("CarInStock", () => {
    const tree = renderer.create(<CarInStock available={true} />);
    expect(tree).toMatchSnapshot();
  });
  it("DisplayCarsGrid", () => {
    const tree = renderer.create(<DisplayCarsGrid />);
    expect(tree).toMatchSnapshot();
  });
  it("FormWrapper", () => {
    const tree = renderer.create(<FormWrapper />);
    expect(tree).toMatchSnapshot();
  });
  it("styledSelect", () => {
    const tree = renderer.create(<StyledSelect value={""} />);
    expect(tree).toMatchSnapshot();
  });
  it("Row", () => {
    const tree = renderer.create(<Row />);
    expect(tree).toMatchSnapshot();
  });
});
