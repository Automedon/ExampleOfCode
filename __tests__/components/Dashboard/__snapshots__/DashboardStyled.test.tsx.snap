// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled-components Dashboard CarInStock 1`] = `
<div
  className=""
/>
`;

exports[`Styled-components Dashboard DisplayCarsGrid 1`] = `
<div
  className=""
/>
`;

exports[`Styled-components Dashboard FormWrapper 1`] = `
<div
  className=""
/>
`;

exports[`Styled-components Dashboard Row 1`] = `
<div
  className=""
/>
`;

exports[`Styled-components Dashboard styledSelect 1`] = `
<div
  className="MuiInputBase-root MuiInput-root MuiInput-underline "
  onClick={[Function]}
>
  <div
    aria-haspopup="listbox"
    className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "&#8203;",
        }
      }
    />
  </div>
  <input
    onAnimationStart={[Function]}
    type="hidden"
    value=""
  />
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSelect-icon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
</div>
`;
